app/views/location.scala.html

```diff
@@ -17,27 +17,38 @@
 @import controllers.routes
 @import forms.Location
 @import views.Title
+@import models.requests.JourneyRequest
+@import components.gds.exportsInputText
+@import components.{errorSummary, sectionHeader}
+@import uk.gov.hmrc.govukfrontend.views.html.components._
 
-@this(main_template: views.html.main_template)
+@this(  govukLayout: views.html.gds_main_template,
+govukButton: GovukButton,
+govukInput: GovukInput,
+exportsInputText: exportsInputText,
+errorSummary: errorSummary,
+sectionHeader: sectionHeader,
+formHelper: uk.gov.hmrc.play.views.html.helpers.FormWithCSRF)
 
-@(form: Form[Location])(implicit request: Request[_], messages: Messages)
+@(form: Form[Location], consignmentReference: Option[String])(implicit request: JourneyRequest[_], messages: Messages)
 
-@main_template(title = Title("location.question")) {
-    @helper.form(routes.LocationController.saveLocation(), 'autoComplete -> "off") {
-        @helper.CSRF.formField
+@govukLayout(
+    title = Title("location.question"),
+    backButtonTitle = messages("site.back"),
+    backButtonUrl = controllers.routes.MovementDetailsController.displayPage.url) {
+    @formHelper(action = controllers.routes.LocationController.saveLocation(), 'autoComplete -> "off") {
 
-        @components.back_link(routes.MovementDetailsController.displayPage())
+        @errorSummary(form.errors)
 
-        @components.error_summary(form.errors)
+        @sectionHeader(messages("movement.sectionHeading", request.answers.`type`.toString.toLowerCase.capitalize, consignmentReference.getOrElse("")))
```

Comment:
Patryk doesn't like capitalisation in message keys (I know you haven't changed this but....)